Store feature flags in database

ddohler opened the pull request:
## Overview

In order to have different feature flags enabled on production and staging, we need to store feature flags in the database rather than in our Scala code. This moves feature flags into a database table and modifies the existing `config` endpoint to return feature flags from that table.

### Checklist

- [ ] ~Styleguide updated, if necessary~
- [ ] ~Swagger specification updated, if necessary~
- [x] Symlinks from new migrations present or corrected for any new migrations

### Notes

Although the `key` of each feature flag is unique and must remain so in order to avoid confusing the front-end, I included a UUID primary key in the `feature_flags` table anyway. Adding the extra field doesn't cause any problems on the front-end, and it seems like just about every time we think we can use a human-readable string as a database key, we eventually end up discovering that we're wrong for one reason or another, so I figured adding a UUID was the safest choice.

I did not create a traditional set of REST endpoints for the FeatureFlag table since it didn't seem like we really wanted or needed any.

PR templates ftw! I totally would not have known to update the auto-generated symlink if there hadn't been a note for it.

## Testing Instructions

 * Follow the README instructions to run migrations.
 * Start up the server, navigate to the UI and inspect the result from the `config` endpoint. Confirm that it includes the same `featureFlags` that we had before, and that they now include a UUID `id` key.
 * Open a DB shell with `./scripts/psql`
 * Run `update feature_flags set active = FALSE where key = 'tools-ui';` and then reload the front-end; the Tools UI elements should disappear. Reverse the change and they should reappear.
 * If possible, double-check that I haven't broken any intercom functionality (I had conflicts with that PR which I'm pretty sure I resolved but it wouldn't be bad to double-check).

Closes #1305

alkamin commented:
You may have gotten pre-empted by another PR with migrations.

alkamin commented on app-backend/migrations/src/main/scala/migrations/51.scala:
```diff
@@ -0,0 +1 @@
+../../../../src_migrations/main/scala/50.scala
```
This needs to point to 51.scala, or 52 if you choose to bump your migrations up a number

ddohler replied:
Oops, good catch! Yeah, I noticed the migration numbering conflict; will fix that up.

ddohler commented:
Okay, I've rebased on develop and updated the migration numbers. If you've run the pre-rebase migrations you'll need to run `TRUNCATE TABLE feature_flags;` in order to test the new ones.
